### PR TITLE
Build for macOS with ubuntu runner, support 64-bit ARM architecture on Windows and Linux

### PR DESCRIPTION
**Pull request by naotama2002**
###  Build for macOS with ubuntu runner

When cross-compiling for macOS, DNS resolution was handled improperly. This has been fixed by golang in version 1.20. Build on Ubuntu runner now that the problem is fixed.

#### 🔗 References
- https://danp.net/posts/macos-dns-change-in-go-1-20/
- https://github.com/golang/go/issues/12524
- https://go-review.googlesource.com/c/go/+/446178

### Support 64-bit ARM architecture
- Windows
- Linux

## ✅ Test

- Run on macOS
    - [x] ARM64
    - [x] AMD64
- [x] Run on Windows; ARM64
- [x] Run on Linux; ARM64

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
   release:
     # Run the build on macOS, because only the macOS version is built with CGO enabled.
     # See https://github.com/golang/go/issues/12524
```
Comment (korosuke613) [inline .github/workflows/release.yml:11]: I think this comment is no longer necessary.

**Review by korosuke613 — CHANGES_REQUESTED** [review]
`.github/workflows/ci.yml` also uses macos-13 runner, so please use ubuntu-latest here as well.

https://github.com/cybozu/assam/blob/bfa89612e1f083210be380f3e39808dcfffb010d/.github/workflows/ci.yml#L50-L52

---

Also, please check additional comments.
For the rest, it is LGTM 👍🏼

File: .github/workflows/release.yml (hunk 1 above)
Comment (naotama2002) [reply .github/workflows/release.yml:11]: Think you for review.

Package execution on CI is also changed to be performed by ubuntu runner.
https://github.com/cybozu/assam/pull/263/commits/0611838f8817e778a66cc1b2be6f6885178de538

Comment (korosuke613) [reply .github/workflows/release.yml:11]: Thanks! I think ci.yaml is good.

However, `release.yml` contains `Run the build on ...` comments are still there. Please check it.

Comment (naotama2002) [reply .github/workflows/release.yml]: Thank you for review.

Delete coment ```Run the build on ...``` on release.yml.
https://github.com/cybozu/assam/pull/263/commits/fe2bc1d55d5763cdeec8ad53b8d8867782aa24d3#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34L10-L12

**Review by korosuke613 — APPROVED** [review]
LGTM!

**Pull request merged**